### ISSUE TITLE
mpdstats: no rating when the same song is repeated

### ISSUE THREAD

**Issue by ray66** [beetbox/beets#2753]
### Problem

When the same song is played several times only the last playback is rated. This happens

- when there is only one song in the playlist and repeat mode is on
- when the playlist contains the same song twice in sequence 

### Setup

* OS: Raspbian Stretch
* Python version 2.7.13
* beets version 1.4.5

**Comment by sampsyo** [comment]
Indeed. I think this is because the current strategy for detecting when a song has been played is by checking when a new track is playing since we last looked.

Any ideas for an alternative way to detect song completion? We could check for when the current playback time goes backward, but this might erroneously detect multiple plays when the user is just scrubbing around in the song.

**Comment by ray66** [comment]
One solution could be 

- when the song starts, set a timer close to the expected end of the song, for example 2 seconds before.
- when the timer elapses before the song changed, consider the song as played
- otherwise the song is considered as skipped

This would require to wait for MPD events and a timer the same time. I think the present issue is not worth such a major change.

**Comment by sampsyo** [comment]
That's a good point. Let's keep this suggestion here—it is nice to track this defect, and if it becomes more urgent in the future, we can consider this solution (or looking for something easier to implement).

**Comment by gdtwst** [comment]
My commit at #2707 should have fixed this. I'm not sure if it's the best implementation, but I haven't had any issues with ratings not updating when playing the same song repeatedly.

**Comment by sampsyo** [comment]
Right; that’s a good point. Can you try out the latest source, @ray66?
